devtools/forcefield-scripts/amber/amber2omm.py

```diff
@@ -0,0 +1,844 @@
+    tleap_path = find_executable('tleap')
+    AMBERHOME = os.path.split(tleap_path)[0]
+    AMBERHOME = os.path.join(AMBERHOME, '../')
+    parmed.amber.AMBERHOME = AMBERHOME
+
+# set global defaults for verbose and log
+verbose = False
+no_log = False
+
+def main():
+    global verbose
+    global no_log
+    global logger
+    # argparse
+    parser = argparse.ArgumentParser(description='AMBER --> OpenMM forcefield '
+                                                 'conversion script')
+    parser.add_argument('--input', '-i', default='files/master.yaml',
+                        help='path of the input file. Default: "files/master.yaml"')
+    parser.add_argument('--input-format', '-if', default='yaml',
+                        help='format of the input file: "yaml" or "leaprc". Default: "yaml"')
+    parser.add_argument('--verbose', '-v', action='store_true',
+                        help='turns verbosity on')
+    parser.add_argument('--no-log', action='store_true',
+                        help='turns logging of energies to log.csv off')
+    parser.add_argument('--protein-test', action='store_true',
+                        help='validate resulting XML through protein tests')
+    parser.add_argument('--nucleic-test', action='store_true',
+                        help='validate resulting XML through nucleic acid tests')
+    parser.add_argument('--protein-ua-test', action='store_true',
+                        help='validate resulting XML through united-atom protein tests')
+    parser.add_argument('--phospho-protein-test', action='store_true',
+                        help='validate resulting XML through phosphorylated protein tests')
+    parser.add_argument('--gaff-test', action='store_true',
+                        help='validate resulting XML through small-molecule (GAFF) test')
+    # parser.add_argument('--write_unused', action='store_true')
+    # parser.add_argument('--default-warnings', action='store_true')
+    args = parser.parse_args()
+    verbose = args.verbose
+    no_log = args.no_log
+
+    if not no_log: logger = Logger('log.csv')
+
+    # input is either a YAML or a leaprc - default is leaprc
+    # output directory hardcoded here for ffxml/
+    if args.input_format == 'yaml':
+        convert_yaml(args.input, ffxml_dir='ffxml/')
+    # if leaprc converted - output to the same dir
+    elif args.input_format == 'leaprc':
+        ffxml_name = convert_leaprc(args.input, ffxml_dir='./')
+        if args.protein_test:
+            validate_protein(ffxml_name, args.input)
+        if args.nucleic_test:
+            validate_nucleic(ffxml_name, args.input)
+        if args.protein_ua_test:
+            validate_protein(ffxml_name, args.input, united_atom=True)
+        if args.phospho_protein_test:
+            validate_phospho_protein(ffxml_name, args.input)
+        if args.gaff_test:
+            validate_gaff(ffxml_name, args.input)
+    else:
+        raise RuntimeError('Wrong input_format chosen.')
```

Comment:
To me `RuntimeError` is something that indicates some kind of internal inconsistency (same with `AssertionError`.  Since this is being done inside `main()` and isn't going to be caught, the end result of this statement is a hard system exit with a uselessly verbose traceback.  Handling this condition would be much cleaner by replacing it with a simple `sys.exit('Wrong input_format chosen')` (which still exits with a non-zero exit status, but doesn't clutter the output with a traceback).

Basically, IMO you should use `sys.exit` where you intend for an error to quit execution.  Raise an exception when you really want a Traceback.